File: Payload_Type/go_services/command_augment/agentfunctions/builder.go
package agentfunctions

import (
	agentstructs "github.com/MythicMeta/MythicContainer/agent_structs"
	"path/filepath"
)

var payloadDefinition = agentstructs.PayloadType{
	Name:          "MyCustomCommands",
	Author:        "@its_a_feature_",
	SupportedOS:   []string{agentstructs.SUPPORTED_OS_LINUX, agentstructs.SUPPORTED_OS_MACOS},
	Description:   "Extra commands I want added to all agents",
	AgentType:     agentstructs.AgentTypeCommandAugment,
	MessageFormat: agentstructs.MessageFormatJSON,
}

func Initialize() {
	agentstructs.AllPayloadData.Get("MyCustomCommands").AddPayloadDefinition(payloadDefinition)
	agentstructs.AllPayloadData.Get("MyCustomCommands").AddIcon(filepath.Join(".", "command_augment", "agentfunctions", "myCustomCommands.svg"))
}

File: Payload_Type/python_services/basic_command_augment/agent_functions/builder.py
from mythic_container.PayloadBuilder import *
from mythic_container.MythicCommandBase import *


class BasicCommandAugment(PayloadType):
    name = "basic_command_augment"
    author = "@its_a_feature_"
    supported_os = [SupportedOS.MacOS]
    description = """This adds custom jxa functions to all supported agents."""
    agent_path = pathlib.Path(".") / "basic_command_augment"
    agent_icon_path = agent_path / "agent_functions" / "basic_command_augment.svg"
    agent_code_path = agent_path / "agent_code"
    command_augment_supported_agents = [] # this means these commands are injected into all agent's callbacks on macOS
    agent_type = AgentType.CommandAugment
